src/components/Header/Header.tsx

```diff
@@ -0,0 +1,86 @@
+import classNames from 'classnames';
+import React, { useEffect, useMemo, useRef } from 'react';
+import { Todo } from '../../types/Todo';
+import { USER_ID } from '../../api/todos';
+
+type Props = {
+  title: string;
+  setTitle: (title: string) => void;
+  errorMessage: string;
+  setErrorMessage: (message: string) => void;
+  resetError: () => void;
+  addTodo: (todo: Omit<Todo, 'id'>) => Promise<void>;
+  completed: boolean;
+  todos: Todo[];
+  submitting: boolean;
+};
+
+export const Header: React.FC<Props> = ({
+  title,
+  setTitle,
+  errorMessage,
+  setErrorMessage,
+  resetError,
+  addTodo,
+  completed,
+  todos,
+  submitting,
+}) => {
+  const formField = useRef<HTMLInputElement>(null);
+
+  function handleTitleChange(event: React.ChangeEvent<HTMLInputElement>) {
```

Comment:
Consider will be better if you use function expression instead of function declaration, fix it everywhere
```suggestion
  comst handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```